### PR TITLE
x11-misc/gtkdialog: revbump, fix build for clang16

### PR DESCRIPTION
**Pull request by Schievel1**
I don't know what sense it makes to have a live ebuild when the last commit on upstreams repo was 9 years ago and the _p is from 2020, but well... I also overhauled the live ebuild. 
So essentially both, the live ebuild and the non-live ebuild, compile the same code anyway. 

@thesamesam take a closer look at that patch, I will mark the things I don't feel very confident about 

Closes: https://bugs.gentoo.org/875704
Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>

### CODE REVIEW

File: x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+ 
+ extern gchar *option_include_file;
+ 
+--- a/src/automaton.c
++++ b/src/automaton.c
+@@ -35,6 +35,7 @@
+ */
+ 
+ #include <gtk/gtk.h>
++#include <glib/gprintf.h>
```
Comment (Schievel1) [inline x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch:25]: Can I just import glib like this? Without the ebuild depending on it?

File: x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+-instruction_execute(instruction command)
++int instruction_execute(instruction command)
+ {
+ 	GList *element;
+ 	token Token;
+@@ -911,9 +912,11 @@ gboolean widget_moved(GtkWidget *widget,
+ 		//gtk_widget_set_usize(window, 
+ 		//		configure->width -20,
+ 		//		configure->height);
++#if HAVE_GTK==2
```
Comment (Schievel1) [inline x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch:62]: gtk_widget_set_uposition is defined in `/usr/include/gtk-2.0/gtk-gtkwidget.h`. 
If gtkdialog is configured to use gtk3 (what we hardcode in the ebulid) we can not assume it is there and include that file. 
That is why I put that pp-if there. But does this work this way? HAVE_GTK is defined in configure.

File: x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
++extern void print_program();
++extern int gtkdialog_lex (void);
++extern int gtkdialog_parse (void);
+--- a/src/printing.c
++++ b/src/printing.c
+@@ -25,6 +25,7 @@
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <gtk/gtk.h>
++#include <glib/gprintf.h>
```
Comment (Schievel1) [inline x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch:96]: Can I just import glib like this? Without the ebuild depending on it?

File: x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
++++ b/src/gtkdialog_parser.h
+@@ -385,3 +385,8 @@ typedef union YYSTYPE
+ extern YYSTYPE gtkdialog_lval;
+ 
+ 
++extern int gtkdialog_error(char *c);
++extern void run_program(void);
++extern void print_program();
++extern int gtkdialog_lex (void);
++extern int gtkdialog_parse (void);
```
Comment (Schievel1) [inline x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch:89]: I could not find that function anywhere in the codebase. I just made that prototype up. It somehow works. :question: :question: :question: :question:

**Comment by Schievel1** [comment]
I just noticed this is triggering pointer warnings, so back to draft.

File: x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch (hunk 3 above)
Comment (thesamesam) [reply x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch:96]: No, the include will fail if glib isn't installed

File: x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch (hunk 4 above)
Comment (thesamesam) [reply x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch:89]: Yeah this is one of those yacc things and I think it's fine.

File: x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch (hunk 2 above)
Comment (thesamesam) [reply x11-misc/gtkdialog/files/gtkdialog-0.8.3-fix-build-for-clang16.patch:62]: I think this looks fine.

**Pull request closed without merging**